sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -138,6 +141,39 @@ def main(unused_argv):
       fn_log_handler.close()
 
 
+def _get_worker_count(pipeline_options):
+  """Extract worker count from the pipeline_options.
+
+  This defines how many SdkWorkers will be started in this Python process.
+  And each SdkWorker will have its own thread to process data. Name of the
+  experimental parameter is 'worker_threads'
+  Example Usage in the Command Line:
+    --experimental worker_threads=1
+
+  Note: worker_threads is an experimental flag and might not be available in
+  future releases.
+
+  Returns:
+    an int containing the worker_threads to use. Default is 1
+  """
+  pipeline_options = pipeline_options.get(
+      'options') if pipeline_options.has_key('options') else {}
+  experiments = pipeline_options.get(
+      'experiments'
+  ) if pipeline_options and pipeline_options.has_key('experiments') else []
+
+  experiments = experiments if experiments else []
+
+  for experiment in experiments:
+    # There should only be 1 match so returning from the loop
+    if re.match(r'worker_threads=', experiment):
+      return int(
+          re.match(r'worker_threads=(?P<worker_threads>.*)',
+                   experiment).group('worker_threads'))
+
+  return 1
```

Comment:
<!--new_thread; commit:aca5c59424a44e3245d6a4c5ff311dad7eee673c; resolved:0-->
I prefer a default of at least 2, so that we can understand the dynamics of threading. I think we will eventually settle on a even a higher default value once do some benchmarking.

If you are not comfortable with this yet, we can do it in a follow up PR.

Reply:
I would prefer changing the default in a subsequent PR after having concrete reproducible benchmark numbers.